c++/include/orc/Reader.hh

```diff
@@ -149,6 +149,26 @@ namespace orc {
      */
     RowReaderOptions& includeTypes(const std::list<uint64_t>& types);
 
+    /**
+     * A map of <typeId, ArrayReadIntent> used as a parameter in
+     * RowReaderOptions::includeTypesWithIntents.
+     */
+    typedef std::map<uint64_t, ArrayReadIntent> TypeReadIntents;
+
+    /**
+     * Selects which type ids to read and specific ArrayReadIntents for each
+     * type id. The root type is always 0 and the rest of the types are
+     * labeled in a preorder traversal of the tree. The parent types are
+     * automatically selected, but the children are not.
+     *
+     * This option clears any previous setting of the selected columns or
+     * types.
+     * @param typesWithIntents a map of <typeId, ArrayReadIntent>.
```

Comment:
Please use `TypeReadIntents` instead of `<typeId, ArrayReadIntent>`.

Reply:
Deleted in latest commit.